src/evaluation/utils/custom_pipelines.py

```diff
@@ -203,3 +201,189 @@ def run(self, query, top_k_retriever, top_k_reader, labels):
         )
 
         return output
+
+class HottestReaderPipeline(BaseStandardPipeline):
+    def __init__(
+        self,
+        reader: BaseReader,
+        retriever_title: BaseRetriever,
+        retriever_bm25: BaseRetriever,
+        k_title_retriever: int,
+        k_bm25_retriever: int,
+        threshold_score: float,
+    ):
+        """
+        Initialize a Pipeline for Extractive Question Answering. This Pipeline is based on two retrievers and a reader.
+        The two retrievers used for this pipeline are :
+
+            - A TitleEmbeddingRetriever
+            - An ElasticsearchRetriever
+        The output of the two retrievers are concatenated based on the number of k_retriever passed for each retrievers.
+
+        :param reader: Reader instance :param retriever: Retriever instance :param k_title_retriever: int :param
+        k_bm25_retriever: int
+        """
+        self.k_title_retriever = k_title_retriever
+        self.k_bm25_retriever = k_bm25_retriever
+        self.pipeline = Pipeline()
+        self.pipeline.add_node(
+            component=retriever_bm25, name="Retriever_bm25", inputs=["Query"]
+        )
+        self.pipeline.add_node(
+            component=retriever_title, name="Retriever_title", inputs=["Query"]
+        )
+        self.pipeline.add_node(
+            component=JoinDocuments(ks_retriever=[k_bm25_retriever, k_title_retriever]),
+            name="JoinRetrieverResults",
+            inputs=["Retriever_bm25", "Retriever_title"],
+        )
+        self.pipeline.add_node(
+            component=reader, name="QAReader", inputs=["JoinRetrieverResults"]
+        )
+        self.pipeline.add_node(
+            component=AnswerifyDocuments(), name="AnswerFromRetrievers", inputs=["JoinRetrieverResults"]
+        )
+        self.pipeline.add_node(
+            component=JoinAnswers(threshold_score=threshold_score),
+            name="JoinResults",
+            inputs=["Query","QAReader", "AnswerFromRetrievers"],
+        )
+
+    def run(
+        self,
+        query: str,
+        filters: Optional[Dict] = None,
+        top_k_retriever: int = 10,
+        top_k_reader: int = 10,
+    ):
+        assert top_k_retriever <= max(
+            self.k_title_retriever, self.k_bm25_retriever
+        ), "Be carefull, the pipeline was run with top_k_retriever that is greater than the k_retriever declared at instanciation"
+        output = self.pipeline.run(
+            query=query,
+            filters=filters,
+            top_k_retriever=top_k_retriever,
+            top_k_reader=top_k_reader,
+        )
+        return output
+
+
+class AnswerifyDocuments(BaseComponent):
```

Comment:
The classes defined below could be imported from `custom_component.py` to avoid redundancy in the code base ?

Reply:
What do you think @Rob192 ?

Reply:
yes, this is what we discussed yesterday, I need to delete them from this file

Reply:
ok. I approved it already, so you'll be able to merge once you've done that !

Reply:
I just added a property in the meta so that "answerfied" answers can be easily identified by the frontend :)

Reply:
ok cool